src/user/user.service.ts

```diff
@@ -67,12 +72,19 @@ export class UserService {
     })
   }
 
-  async findAllUsers(): Promise<User[]> {
-    return await this.userDatabaseService.findAllUsers()
+  async getAllUsers(): Promise<User[]> {
+    return await this.userDatabaseService.getAllUsers()
   }
 
-  async findUserByUsername(username: string): Promise<User> {
-    return await this.userDatabaseService.findUserByUsername(username)
+  async getUserById(id: string): Promise<User> {
+    return await this.userDatabaseService.getUserById(id)
+  }
+
+  async getUserTodos(userId: string): Promise<Todo[]> {
+    await this.getUserById(userId)
```

Comment:
```suggestion
```
Redundant call

Reply:
It placed here to check if user founded in `getUserById` method as an easiest way to check it and prevent code duplication since we already have this method with error handling inside:
```
async getUserById(id: string): Promise<User> {
    const userById = await this.userModel.findById(id)

    if (!userById) {
      throw new NotFoundError(`User with id ${id} not found`)
    }

    return userById.toObject()
  }
```
Does it an acceptable way or should I use something else?

Reply:
Why do we need an additional check if user exists? Isn't `todosService.getAllTodosByUserId(userId)` will return an empty array if no todos found for given ID (including cases when user doesn't exist)?

Reply:
Yes, that's true, but won't it create confusion when, for example, a user requests todos for a non-existent user and receives an empty array, continuing to think that this user simply doesn't have todos?

Reply:
Fixed:
```
async getUserTodos(userId: string): Promise<Todo[]> {
    const isUserExist = await this.isUserExistById(userId)

    if (!isUserExist) {
      throw new NotFoundError(`User with id ${userId} not found`)
    }

    return await this.todosService.getAllTodosByUserId(userId)
  }
```